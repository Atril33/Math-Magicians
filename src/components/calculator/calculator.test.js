import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MyCalculator from './calculator';

test('Snapshot test for Calculator.js', () => {
  const comp = renderer.create(
    <MyCalculator />,
  );
  const tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('MyCalculator', () => {
  it('Should be press correct Button', () => {
    const { getByText } = render(<MyCalculator />);

    expect(getByText('AC')).toBeInTheDocument();
    expect(getByText('+/-')).toBeInTheDocument();
    expect(getByText('%')).toBeInTheDocument();
    expect(getByText('รท')).toBeInTheDocument();
    expect(getByText('7')).toBeInTheDocument();
    expect(getByText('8')).toBeInTheDocument();
    expect(getByText('9')).toBeInTheDocument();
    expect(getByText('x')).toBeInTheDocument();
    expect(getByText('4')).toBeInTheDocument();
    expect(getByText('5')).toBeInTheDocument();
    expect(getByText('6')).toBeInTheDocument();
    expect(getByText('-')).toBeInTheDocument();
    expect(getByText('1')).toBeInTheDocument();
    expect(getByText('2')).toBeInTheDocument();
    expect(getByText('3')).toBeInTheDocument();
    expect(getByText('+')).toBeInTheDocument();
    expect(getByText('0')).toBeInTheDocument();
    expect(getByText('.')).toBeInTheDocument();
    expect(getByText('=')).toBeInTheDocument();
  });

  test('Should subtract two numbers correctly by pressing - ', () => {
    const { getByText } = render(<MyCalculator />);
    fireEvent.click(getByText('5'));
    fireEvent.click(getByText('-'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    const result = '3';
    expect(result).toEqual('3');
  });
});
