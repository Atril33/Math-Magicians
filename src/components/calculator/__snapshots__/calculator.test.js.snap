// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Calculator.js 1`] = `
<center>
  <div
    className="container"
  >
    <input
      id="calculator-input"
      readOnly={true}
      type="number"
      value="0"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="divide orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="multply orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="sub orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="plus orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="Zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="isqual orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</center>
`;
